File: keystone-field-service-management/src/main/java/com/keystone/service/SiteService.java
package com.keystone.service;

import java.util.List;

import com.keystone.dto.SiteDTO;

public interface SiteService {

    // Create Site
    SiteDTO createSite(SiteDTO siteDTO);

    // Get All Sites
    List<SiteDTO> getAllSites();

    // Get Site By Id
    SiteDTO getSiteById(Long id);

    // Update Site
    SiteDTO updateSite(Long id, SiteDTO siteDTO);

    // Delete Site
    void deleteSite(Long id);

}

File: keystone-frontend/src/services/SiteService.ts
import api from "../api/axios";
import type { Site } from "../types/Site";

interface ApiResponse<T> {
  timestamp: string;
  status: number;
  success: boolean;
  message: string;
  data: T;
}

// Get All Sites
export const getAllSites = async (): Promise<Site[]> => {
  const response = await api.get<ApiResponse<Site[]>>("/sites");
  return response.data.data;
};

// Get Site By Id
export const getSiteById = async (id: number): Promise<Site> => {
  const response = await api.get<ApiResponse<Site>>(`/sites/${id}`);
  return response.data.data;
};

// Create Site
export const createSite = async (site: Site): Promise<Site> => {
  const response = await api.post<ApiResponse<Site>>("/sites", site);
  return response.data.data;
};

// Update Site
export const updateSite = async (
  id: number,
  site: Site
): Promise<Site> => {
  const response = await api.put<ApiResponse<Site>>(
    `/sites/${id}`,
    site
  );
  return response.data.data;
};

// Delete Site
export const deleteSite = async (id: number): Promise<void> => {
  await api.delete(`/sites/${id}`);
};